contracts/KyberOasisReserve.sol

```diff
@@ -0,0 +1,273 @@
+
+        kyberNetwork = _kyberNetwork;
+        KyberNetworkSet(kyberNetwork);
+    }
+
+    event OtcSet(address otc);
+
+    function setOtc(OtcInterface _otc) public onlyAdmin {
+        require(_otc != address(0));
+
+        wethToken.approve(otc, 0);
+        tradeToken.approve(otc, 0);
+        wethToken.approve(_otc, 2**255);
+        tradeToken.approve(_otc, 2**255);
+
+        otc = _otc;
+        OtcSet(otc);
+    }
+
+    event WethTokenSet(TokenInterface wethToken);
+
+    function setWethToken(TokenInterface _wethToken) public onlyAdmin {
+        require(_wethToken != address(0));
+        require(getDecimals(_wethToken) == MAX_DECIMALS);
+
+        wethToken.approve(otc, 0);
+        _wethToken.approve(otc, 2**255);
+
+        wethToken = _wethToken;
+        WethTokenSet(wethToken);
+    }
+
+    event TradeTokenSet(ERC20 tradeToken);
+
+    function setTradeToken(ERC20 _tradeToken) public onlyAdmin {
+        require(_tradeToken != address(0));
+        require(getDecimals(_tradeToken) == MAX_DECIMALS);
+
+        tradeToken.approve(otc, 0);
+        _tradeToken.approve(otc, 2**255);
+
+        tradeToken = _tradeToken;
+        TradeTokenSet(tradeToken);
+    }
+
+    event FeeBpsSet(uint feeBps);
+
+    function setFeeBps(uint _feeBps) public onlyAdmin {
+        require(_feeBps < 10000);
+
+        feeBps = _feeBps;
+        FeeBpsSet(feeBps);
+    }
+
+    function getDestQty(ERC20 src, ERC20 dest, uint srcQty, uint rate) public view returns(uint) {
+        uint dstDecimals = getDecimals(dest);
+        uint srcDecimals = getDecimals(src);
+
+        return calcDstQty(srcQty, srcDecimals, dstDecimals, rate);
+    }
+
+    function valueAfterReducingFee(uint val) public view returns(uint) {
+        require(val <= MAX_QTY);
+        return ((10000 - feeBps) * val) / 10000;
+    }
+
+    function getConversionRate(ERC20 src, ERC20 dest, uint srcQty, uint blockNumber) public view returns(uint) {
+        uint  rate;
+        uint  destQty;
+        uint  actualSrcQty;
+        ERC20 wrappedSrc;
+        ERC20 wrappedDest;
+
+        blockNumber;
+
+        if (!tradeEnabled) return 0;
+        if ((tradeToken != src) && (tradeToken != dest)) return 0;
+
+        if (src == ETH_TOKEN_ADDRESS) {
+            wrappedSrc = wethToken;
+            wrappedDest = dest;
+            actualSrcQty = srcQty;
+        } else if (dest == ETH_TOKEN_ADDRESS) {
+            wrappedSrc = src;
+            wrappedDest = wethToken;
+
+            if (srcQty < ETH_SRC_AMOUNT) {
+                /* Assuming token is stable, use a minimal amount to get rate also for small token quant. */
+                actualSrcQty = ETH_SRC_AMOUNT;
+            } else {
+                actualSrcQty = srcQty;
+            }
+        } else {
+            return 0;
+        }
+
+        destQty = otc.getBuyAmount(wrappedDest, wrappedSrc, actualSrcQty);
+        rate = calcRateFromQty(actualSrcQty, valueAfterReducingFee(destQty), MAX_DECIMALS, MAX_DECIMALS);
```

Comment:
maybe just define special constant in this contract

Reply:
ok, using COMMON_DECIMALS.